app/v2/controllers/stream/StreamingParsers.scala

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v2.controllers.stream
+
+import akka.stream.IOResult
+import akka.stream.Materializer
+import akka.stream.scaladsl.FileIO
+import akka.stream.scaladsl.RunnableGraph
+import akka.stream.scaladsl.Source
+import akka.util.ByteString
+import play.api.libs.Files
+import play.api.libs.streams.Accumulator
+import play.api.mvc.BaseControllerHelpers
+import play.api.mvc.BodyParser
+import play.api.mvc.Request
+
+import java.util.concurrent.Executors
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+
+trait StreamingParsers {
```

Comment:
Quite a lot of this trait is unnecessary for this project - we only care about `streamFromMemory` here as that's all we'll use, I suggest removing the other options from here.